exp/api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -486,3 +493,183 @@ func (m *AzureManagedControlPlane) validateName(_ client.Client) error {
 
 	return nil
 }
+
+// validateAutoScalerProfile validates an AutoScalerProfile.
+func (m *AzureManagedControlPlane) validateAutoScalerProfile(_ client.Client) error {
+	var allErrs field.ErrorList
+
+	if m.Spec.AutoScalerProfile == nil {
+		return nil
+	}
+
+	if m.Spec.AutoScalerProfile.BalanceSimilarNodeGroups != nil {
```

Comment:
Because we're already enforcing this via the kubebuilder API spec, we don't need to add additional webhook enforcement here

Reply:
removed